test/extensions/filters/listener/http_inspector/http_inspector_test.cc

```diff
@@ -485,6 +553,87 @@ TEST_F(HttpInspectorTest, MultipleReadsHttp1BadProtocol) {
   EXPECT_EQ(1, cfg_->stats().http_not_found_.value());
 }
 
+TEST_F(HttpInspectorTest, Http1WithLargeRequestLine) {
+  init();
+  absl::string_view method = "GET", http = "/index HTTP/1.0\r";
+  std::string spaces(Config::MAX_INSPECT_SIZE - method.length() - http.length(), ' ');
+  const std::string data = absl::StrCat(method, spaces, http);
+  {
+    InSequence s;
+
+    EXPECT_CALL(os_sys_calls_, recv(42, _, _, MSG_PEEK)).WillOnce(InvokeWithoutArgs([]() {
+      return Api::SysCallSizeResult{ssize_t(-1), EAGAIN};
+    }));
+
+    uint64_t num_loops = Config::MAX_INSPECT_SIZE;
+#if defined(__has_feature) &&                                                                      \
+    ((__has_feature(thread_sanitizer)) || (__has_feature(address_sanitizer)))
+    num_loops = 2;
+#endif
+
+    for (size_t i = 1; i <= num_loops; i++) {
+      ssize_t len = i;
+      if (num_loops == 2) {
+        len = Config::MAX_INSPECT_SIZE / (3 - i);
+      }
+      EXPECT_CALL(os_sys_calls_, recv(42, _, _, MSG_PEEK))
+          .WillOnce(
+              Invoke([&data, len](int, void* buffer, size_t length, int) -> Api::SysCallSizeResult {
+                ASSERT(length >= data.size());
+                memcpy(buffer, data.data(), data.size());
```

Comment:
`len` instead of `data.size()`? (ditto for below)

Reply:
@yxue I meant `len` here exactly 😅to copy less memory, not `size()` vs `length()`.

Reply:
😳